v4/export/writer_util.go

```diff
@@ -114,6 +121,37 @@ func (b *writerPipe) ShouldSwitchStatement() bool {
 		(b.statementSizeLimit != UnspecifiedSize && b.currentStatementSize >= b.statementSizeLimit)
 }
 
+// SpeedLimiter used for control speed of dump data
+type SpeedLimiter struct {
+	limiter *rate.Limiter
+	limit  uint64
+}
+
+// CheckSpeed check current speed of dump, if already over, automatically suspended.
+func (sl *SpeedLimiter) CheckSpeed(pCtx *tcontext.Context, newSize uint64) {
+	// Less than or equal to 0 no speed limit.
+	if sl.limit <= 0 {
+		return
+	}
+
+	// The number of bytes currently downloaded has exceeded the speed limit value and needs to be paused.
+	err := sl.limiter.WaitN(pCtx, int(newSize))
+	if err != nil {
+		pCtx.L().Error("fail to speed limit.", zap.Error(err))
+	}
+}
+
+// SpeedLimiter constructor methods
+func NewSpeedLimiter(limit int) *SpeedLimiter {
+	limitMB := uint64(limit * 1024 * 1024)
+	limiter := rate.NewLimiter(rate.Limit(limitMB), int(limitMB))
```

Comment:
> It returns an error if n exceeds the Limiter's burst size

This paragraph is copied from "golang.org/x/time".
We should make sure this burst size is larger than any `AddFileSize` that we meet.

Reply:
Add sliding window inserts to prevent oversize.